sway-core/src/ir_generation/const_eval.rs

```diff
@@ -188,7 +197,14 @@ pub(super) fn compile_constant_expression(
         module_ns,
         function_compiler,
         const_expr,
-    )?;
+    )
+    .map_err(|err| match err {
+        CompileError::CannotBeEvaluatedToConst { .. } => CompileError::NonConstantDeclValue {
+            span: call_path.span(),
```

Comment:
This gives an impression these two errors are actually the same. Am I missing something? (Usually we `map_err` to lift e.g. parser error into a compiler error or something like that).

Reply:
I thought about making `CannotBeEvaluatedToConst` local to `const_eval.rs`. I can make it this way if you think more sense. For example:

```rust
enum ConstEvalError {
    CompileError(CompileError),
    CannotBeEvaluatedToConst { span: Span}
}
```

Reply:
I think this approach with local errors specific to const evaluation is better, given that it can be extended to more informative error messages in the future, i.e. we could explain better why some initializer fails to be evaluated

Reply:
Done.